Tasks/VsTest/vstest.ts

```diff
@@ -0,0 +1,515 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('vsts-task-lib/task');
+import path = require('path');
+import Q = require('q');
+var os = require('os');
+var regedit = require('regedit');
+var uuid = require('node-uuid');
+var fs = require('fs');
+var xml2js = require('xml2js');
+
+try {
+    var vsTestVersion: string = tl.getInput('vsTestVersion');
+    var testAssembly: string = tl.getInput('testAssembly', true);
+    var testFiltercriteria: string = tl.getInput('testFiltercriteria');
+    var runSettingsFile: string = tl.getPathInput('runSettingsFile');
+    var codeCoverageEnabled: boolean = tl.getBoolInput('codeCoverageEnabled');
+    var pathtoCustomTestAdapters: string = tl.getInput('pathtoCustomTestAdapters');
+    var overrideTestrunParameters: string = tl.getInput('overrideTestrunParameters');
+    var otherConsoleOptions: string = tl.getInput('otherConsoleOptions');
+    var testRunTitle: string = tl.getInput('testRunTitle');
+    var platform: string = tl.getInput('platform');
+    var configuration: string = tl.getInput('configuration');
+    var publishRunAttachments: string = tl.getInput('publishRunAttachments');
+    var runInParallel: boolean = tl.getBoolInput('runInParallel');
+
+    tl._writeLine("##vso[task.logissue type=warning;TaskName=VSTest]");
+
+    var sourcesDirectory = tl.getVariable('System.DefaultWorkingDirectory');
+    var testAssemblyFiles = getTestAssemblies();
+
+    if (testAssemblyFiles && testAssemblyFiles.size != 0) {
+        var workingDirectory = path.join(sourcesDirectory, "..");
+        getTestResultsDirectory(runSettingsFile, path.join(workingDirectory, 'TestResults')).then(function(resultsDirectory) {
+            invokeVSTest(resultsDirectory).then(function(code) {
+                try {
+                    publishTestResults(resultsDirectory);
+                    tl.setResult(code, tl.loc('VstestReturnCode', code));
+                }
+                catch (error) {
+                    tl._writeLine("##vso[task.logissue type=error;code=" + error + ";TaskName=VSTest]");
+                    throw error;
+                }
+            })
+                .fail(function(err) {
+                    tl._writeLine("##vso[task.logissue type=error;code=" + err + ";TaskName=VSTest]");
+                    throw err;
+                });
+        });
+    }
+    else {
+        tl._writeLine("##vso[task.logissue type=warning;code=002004;]");
+        tl.warning("No test assemblies found matching the pattern: " + testAssembly);
+    }
+}
+catch (error) {
+    tl._writeLine("##vso[task.logissue type=error;code=" + error + ";TaskName=VSTest]");
+    throw error;
+}
+
+function getTestAssemblies(): Set<string> {
```

Comment:
This entire function is what tool.match does it for you out of the box. You don't have to call exclude/include seperately

https://github.com/isaacs/minimatch